### DISCUSSION TITLE
📝 [RFC] Support aria-* in closable

### DISCUSSION THREAD

**Discussion by kiner-tang** [ant-design/ant-design#47188] [RFCs]
## Summary

Sometimes, we need to add `aria-*` attributes to the close button of closable components to support accessibility. Therefore, we need to expand the `closable` type to support the configuration of `aria-*` attributes.

## Motivation

As mentioned in [allow change close aria-label ](https://github.com/react-component/dialog/pull/385) we need to support the configuration of `aria-*` attributes for the close button.

The components that need to be supported are as follows:

- [x] `Alert` #47474
- [x] `Drawer` #47543
- [x] ~~`Message`~~
- [x] `Modal` #47650
- [x] `Notification` #47710
- [x] `Tag` #47678
- [x] `Tour` https://github.com/react-component/tour/pull/52

Support for static methods:

- [x] ~~`Message`~~
- [x] `Modal`
- [x] `Notification`

Some components need to be linked to transform `rc-*`, such as: `Modal`.

## API

Type: ``closable: boolean | { [`aria-${string}`]: string, icon?: React.ReactNode }``

> `closeIcon` 作为语法糖不废弃

## Basic Example

```tsx
<Modal
  title="Basic Modal"
  open={isModalOpen}
  onOk={handleOk}
  onCancel={handleCancel}
  closable={{
    'aria-label': 'this is test aria label',
  }}
>
  <p>Some contents...</p>
  <p>Some contents...</p>
  <p>Some contents...</p>
</Modal>

```

**Comment by kiner-tang** [comment]
@zombieJ @MadCcc 我记得文档之前是故意把 `closable` 属性隐藏的，推荐用户使用 `closeIcon`，这次要不要再文档里面加回去呢？

**Reply by afc163** [reply]
推荐用户使用 `closable={{ closeIcon: xx }}`，保持对 closeIcon 的兼容就行。

**Reply by kiner-tang** [reply]
那也就是说，咱们文档里面把 closeIcon 去掉，并在文档中把 closable 加回去，然后对 closeIcon 是否需要走废弃流程呢？

**Reply by afc163** [reply]
废弃吧，closable 应该是更标准的推荐用法。

**Reply by afc163** [reply]
可以修改一下 https://github.com/ant-design/ant-design/wiki/API-Naming-rules#reference

**Reply by zombieJ** [reply]
会上的结论是 `closeIcon` 作为 `closable` 的语法糖，因为配置无障碍属性相对改 Icon 是个低频需求。多套一层用起来是比较麻烦的。不适合 warning 用户。
